app/styles/_state.scss

```diff
@@ -1,3 +1,84 @@
+.error,
+.success {
+  border-radius: $small-border-radius;
+  color: $message-text-color;
+  display: none;
+  margin-bottom: 5px;
+  padding: 5px;
+  position: relative;
+  top: -15px;
+
+  a {
+    color: $message-text-color;
+  }
+}
+
+.error {
+  background: $error-background-color;
+
+  &.visible {
+    display: block;
+  }
+}
+
+.success {
+  background: $success-background-color;
+  display: none;
+}
+
+// complete is like success, but without the associated styles.
+.complete {
+  display: none;
+}
+
+.progress-hidden {
+  display: none;
+}
+
+.hidden {
+  display: none;
+}
+
+.reset-warning {
+  font-size: $small-font;
+  font-weight: normal;
+}
+
+// Put a margin on top when the error messages follow a div
+div + .success,
+div + .error {
+  margin-top: 10px;
+}
+
+/* Animation triggering classes */
```

Comment:
I'm not 100% on your naming structures, or if it's worth abstracting all these glorious animations into a separate module like _modules/_animation.scss_ to keep them separated but together.